### ISSUE TITLE
Automatically save all queries

### ISSUE THREAD

**Issue by vanatteveldt** [amcat/amcat#576]
AmCAT allows for saving queries, but queries are only saved when a user chooses to do so. 
However, queries are implicitly saved as tasks, since every query is executed as a task, and hence gets a database entry. 

Why not  automatically save every query? The corresponding task would then by a trivial "run_query(123456)" (and maybe we should even have a separate handler for queries and take them away from the normal task list?)

Advantages:
+ Allows a sort of "query history" page, where all queries per name can be seen. By default one entry for each name can be listed to avoid clutter. 
+ Allows more profiling, debugging, etc, especially if we also store some data about the result with the query (e.g. just status, finish_timestamp (->walltime), processing_time)
+ Allows users to share queries by just copy-pasting the address bar (with a nice "share" button for the challenged). (it seems that you can change the URL without reloading? http://spoiledmilk.com/blog/html5-changing-the-browser-url-without-refreshing-page/)
+ Allow clicking to articles to maintain context, e.g. project/123/query/456/article/1, which should make it easier to highlight the article (since the query is known) and allow 'next'/'previous' buttons to read through matching articles. 

Questions:
+ Should we implement this in 3.5 (with addons such as query history etc. potentially moving to 3.5.1?)
+ id or UUID? 
+ Are there any disadvantages that I'm not seeing? :)

**Comment by martijnbastiaan** [comment]
> Allows a sort of "query history" page, where all queries per name can be seen. By default one entry for each name can be listed to avoid clutter.

We can already do this, right?

> Allow clicking to articles to maintain context, e.g. project/123/query/456/article/1, which should make it easier to highlight the article (since the query is known) and allow 'next'/'previous' buttons to read through matching articles.

That's definitely fair, but we don't need to save each query (only tasks).

> Allows users to share queries by just copy-pasting the address bar (with a nice "share" button for the challenged). (it seems that you can change the URL without reloading? http://spoiledmilk.com/blog/html5-changing-the-browser-url-without-refreshing-page/)

You can, but wouldn't this be maddening? This is akin to having git commit every change you make automatically. If I had to guess, 90% of the queries are just for testing things / exploring the dataset. You often want to rollback to some base point you explicitly defined (a saved query), not?

Furthermore, how are you going to handle two users working exploring a saved query at the same time? Are you going to show "the latest" version (whatever that is) in the query list or just the base one where a user explicitly saved the query? Will you develop an interface similar to sharelatex / google docs where you can see which user changed what and allow the user to jump between revisions / rollback?

In short: I think we can implement everything you want with tasks without having to develop complex interfaces.

**Comment by vanatteveldt** [comment]
(closing due to inaction)
